client/utils/base64.c

```diff
@@ -0,0 +1,116 @@
+/*********************************************************************
+* Filename:   base64.c
+* Author:     peter Liu
+* Copyright:
+* Disclaimer: This code is presented "as is" without any guarantees.
+* Details:
+
+*********************************************************************/
+
+
+
+/*************************** HEADER FILES ***************************/
+
+
+#include "base64.h"
+
+/****************************** MACROS ******************************/
+
+
+#define B64_WS              0xE0
+#define B64_ERROR           0xFF
+#define B64_NOT_BASE64(a)   (((a)|0x13) == 0xF3)
+
+#define conv_bin2ascii(a)   (data_bin2ascii[(a)&0x3f])
+#define conv_ascii2bin(a)   (data_ascii2bin[(a)&0x7f])
+
+static const unsigned char data_bin2ascii[65] =
+"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+static const unsigned char data_ascii2bin[128] =
+{ 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xE0, 0xF0, 0xFF, 0xFF,
+        0xF1, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xE0, 0xFF, 0xFF, 0xFF, 0xFF,
+        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x3E, 0xFF, 0xF2, 0xFF, 0x3F, 0x34,
+        0x35, 0x36, 0x37, 0x38, 0x39, 0x3A, 0x3B, 0x3C, 0x3D, 0xFF, 0xFF, 0xFF,
+        0x00, 0xFF, 0xFF, 0xFF, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+        0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10, 0x11, 0x12, 0x13,
+        0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+        0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F, 0x20, 0x21, 0x22, 0x23, 0x24, 0x25,
+        0x26, 0x27, 0x28, 0x29, 0x2A, 0x2B, 0x2C, 0x2D, 0x2E, 0x2F, 0x30, 0x31,
+        0x32, 0x33, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, };
+
+
+/* Base64 Encode */
+int encode_base64(unsigned char *t, const unsigned char *f, int dlen)
```

Comment:
Please, give descriptive names to the parameters or write comment what is **t** and **f**. One-letter parameters create a puzzle.